toolbox/siestabarriers/Utils/flos/neb_analysis.lua

```diff
@@ -0,0 +1,209 @@
+   local na = tonumber(file:read())
+   local R = flos.Array.zeros(na, 3)
+   file:read()
+   local i = 0
+   local function tovector(s)
+      local t = {}
+      s:gsub('%S+', function(n) t[#t+1] = tonumber(n) end)
+      return t
+   end
+   for i = 1, na do
+      local line = file:read()
+      if line == nil then break end
+      -- Get stuff into the R
+      local v = tovector(line)
+      R[i][1] = v[1]
+      R[i][2] = v[2]
+      R[i][3] = v[3]
+   end
+   file:close()
+   return R
+end
+
+-- Now read in the images
+images[1]= flos.MDStep{R=read_geom(image_label .. image_number .. ".xyz")}
+-- Grab the unit table of siesta (it is already created
+-- by SIESTA)
+local Unit = siesta.Units
+
+
+function siesta_comm()
+   -- This routine does exchange of data with SIESTA
+   --local ret_tbl = {}
+     -- Do the actual communication with SIESTA
+  if siesta.state == siesta.INITIALIZE then  --INITIALIZE
+      siesta.receive({"Label","geom.xa","MD.Relaxed"})
+      --siesta.receive({"MD.Relaxed"})
+      --Store the Label
+      label = tostring(siesta.Label)
+      -- Print information
+      siesta_update_DM(image_number)
+      IOprint("\nLUA NEB Analysis for image :".. image_label .. "calculator")  
+      -- This is only reached one time, and that it as the beginning...
+      -- be sure to set the corresponding values
+      siesta.geom.xa = images[1].R* Unit.Ang  --NEB.initial.R * Unit.Ang
+      siesta.send({"geom.xa"})
+      IOprint("\nLUA NEB Analysis for image Coordinates :\n")
+      IOprint(images[1].R* Unit.Ang )
+
+  end
+
+  if siesta.state == siesta.MOVE then
+      -- Here we are doing the actual LBFGS algorithm.
+      -- We retrieve the current coordinates, the forces
+      -- and whether the geometry has relaxed
+      siesta.receive({"MD.Relaxed"})
+      IOprint("\nLUA/NEB MOVE state STARTED\n")
+      siesta.MD.Relaxed = true
+      siesta.send({"MD.Relaxed"})
+  end
+  
+    if siesta.state == siesta.ANALYSIS_AFTER then
+      IOprint("\nLUA/NEB AFTER ANALYSIS state STARTED\n")
+      siesta_update_Analysis(image_number)
+     --os.execute("cp " .. label .. ".PDOS" .. " " .. label .. ".PDOS.3")
+   --   IOprint("\nLUA/NEB MOVE state FINISHED\n")
+
+      siesta_cleanup(cleanup)
+      --ret_tbl = siesta_move(siesta)
+  end
+   
+end
+ 
+function siesta_update_DM(image_number)
+
+   if not siesta.IONode then
+      -- only allow the IOnode to perform stuff...
+      return
+   end
+   -- Move about files so that we re-use old DM files
+   local DM = label .. ".DM"
+   local current_DM = DM .. "." .. tostring(image_number)  
+   -- Saving initial DM
+   IOprint("Saving " .. current_DM .. " to " .. DM)
+   os.execute("cp " .. current_DM .. " " .. DM)   
+   
+end
+
+function siesta_update_Analysis(image_number)
+
+   if not siesta.IONode then
+      -- only allow the IOnode to perform stuff...
+      return
+   end
+   -- Move about files so that we re-use old DM files
+   local PDOS = label .. ".PDOS"
```

Comment:
I really don't think this should be done in `flos` / `lua`. This *must* be the user's responsibility. Consider when new files are added (which are not tracked here), then one would assume this script to do the correct thing, but it doesn't.

Reply:
this comment i don't understand it ?!

Reply:
All the copying of files, it shouldn't be done in this major lua script. If anything it should be a separate thing in flos.

Reply:
ok ill fix this things in flos pr ...